test/http2.js

```diff
@@ -1244,6 +1244,35 @@ test('The h2 pseudo-headers is not included in the headers', async t => {
   t.equal(response.headers[':status'], undefined)
 })
 
+test('HTTP/2 stream closed with NGHTTP2_INTERNAL_ERROR', async (t) => {
+  t.plan(1)
+
+  // Create a new client instance
+  const client = new Client('http://localhost:8000')
```

Comment:
This test fails (you can repro by running `npx tap -t 4 -g NGHTTP2_INTERNAL_ERROR "test/http2.js"`)

This is because the client is trying to hit a non-existent server
```suggestion
  const server = createSecureServer(pem)

  server.on('stream', (stream, headers) => {
    stream.respond({
      ':status': 200,
    })
    stream.end('hello h2!')
  })

  server.listen(0)
  await once(server, 'listening')
  t.teardown(server.close.bind(server))
  
  const client = new Client(`https://localhost:${server.address().port}`, {
  connect: {
    rejectUnauthorized: false,
  },
  allowH2: true,
})
```

The result will be that the test fails with timeout because of the reported issue that this request will never reject or resolve (see other comment)

Reply:
thanks I will update this immediately